internal/api/api.go

```diff
@@ -110,3 +127,46 @@ func versionHeaderMiddleware(version string) func(http.Handler) http.Handler {
 		})
 	}
 }
+
+func sameOriginChecker(logger logr.Logger) func(r *http.Request) bool {
+	return func(r *http.Request) bool {
+		origin := r.Header["Origin"]
+		if len(origin) == 0 {
```

Comment:
I am wondering if this should return false since the client always sets "Origin"? Under what circumstances would this be empty?

Reply:
I believe that browsers should always set origin (see https://stackoverflow.com/a/42242802 for more). Beyond browsers, any other type of client could fake the header or decide not to set it. My understanding is that we should accept them because non-browsers are presumably not affected by CSRF attacks.